lib/instance_billing_flavor_check/utils.py

```diff
@@ -0,0 +1,145 @@
+        try:
+            return config['instance']['dataProvider']
+        except Exception as err:
+            logger.error("Could not parse %s: %s", REGION_SRV_CLIENT_PATH, err)
+    else:
+        logger.error("Could not read file %s", REGION_SRV_CLIENT_PATH)
+    return None
+
+
+
+def get_metadata():
+    command = get_command()
+    if not command:
+        return None
+
+    command = command.split(' ')
+    result = Command.run(command)
+    if result.returncode == 0:
+        return result.output
+    else:
+        logger.error(
+            "Could not fetch the metadata after running the command '%s': '%s' with status '%s'",
+            ' '.join(command),
+            result.error,
+            result.returncode
+        )
+        return None
+
+def get_identifier():
+    try:
+        with open(ETC_OS_RELEASE_PATH) as stream:
+            csv_reader = csv.reader(stream, delimiter="=")
+            os_release = dict(csv_reader)
+    except FileNotFoundError:
+        logger.error("Could not open '%s' file", ETC_RELEASE_PATH)
+        syst.exit(1)
+
+    return os_release.get('NAME')
+
+def get_rmt_ip_addr():
+    """ Return the RMT update server IP the instance is registered to."""
+    try:
+        with open(ETC_HOSTS_PATH) as etc_hosts:
+            etc_hosts_lines = etc_hosts.readlines()
+    except FileNotFoundError:
+        logger.error("Could not open '%s' file", ETC_HOSTS_PATH)
+        syst.exit(1)
+
+    for etc_hosts_line in etc_hosts_lines:
+        if 'susecloud.net' in etc_hosts_line:
+            return etc_hosts_line.split('\t')[0]
+
+
+def make_request(rmtt_ip_addr, metadata, identifier):
+    instance_check_url = f"https://{rmt_ip_addr}/api/instance/check"
+    requests.packages.urllib3.disable_warnings(
+        requests.packages.urllib3.exceptions.InsecureRequestWarning
+    )
+    message = None
+    try:
+        response = requests.get(
+            instance_check_url,
+            timeout=2,
+            verify=False,
+            metadata=metadata,
+            identifer=identifier
+        )
+    except requests.exceptions.HTTPError as err:
+        message = f"Http Error:{err}"
+    except requests.exceptions.ConnectionError as err:
+        message = f"Error Connecting:{err}"
+    except requests.exceptions.Timeout as err:
+        message = f"Timeout Error:{err}"
+    except requests.exceptions.RequestException as err:
+        message = f"Unexpected error:{err}"
+
+    if message:
+        logger.error(message)
+        sys.exit(1)
+    if response.status_code == 200:
+        result = response.json()
+        logger.debug(result)
+        return result.get('state')
+
+    logger.error(
+        'Request to check if instance is PAYG/BYOS failed: %s', response.text
+    )
+
+def check_payg_byos():
+    """Return 'PAYG' OR 'BYOS'."""
+    metadata = get_metadata()
+    identifier = get_identifier()
+    if not metadata or not identifier:
+        sys.exit(1)
```

Comment:
What do we expect SUMa to do when we exit with 1?

Reply:
I do not know if this may be used from script, so inform about exit code looked reasonable 
I do not know how they are going to use it, not problem removing the exit calls

Reply:
I will remove the exit calls then